src/ngraph/op/softmax.cpp

```diff
@@ -28,24 +28,39 @@
 using namespace std;
 using namespace ngraph;
 
+// *** SOFTMAX OP SET 0 ***
 const string op::Softmax::type_name{"Softmax"};
 
-op::Softmax::Softmax(const Output<Node>& arg, const AxisSet& axes)
-    : UnaryElementwiseArithmetic(arg)
+op::v0::Softmax::Softmax(const Output<Node>& arg, const AxisSet& axes)
+    : Op({arg})
     , m_axes(axes)
 {
     constructor_validate_and_infer_types();
 
+    const PartialShape& input_shape = get_input_partial_shape(0);
+    NODE_VALIDATION_CHECK(this,
+                          input_shape.rank().is_static(),
```

Comment:
Not sure if I missed this on first review (or if the original code was just broken), but we need to allow dynamic rank here. Also, these checks belong in `validate_and_infer_types()`, not the ctor.

@mbencer If you like, please feel free to assign a JIRA to me for these issues. I'm working on some written guidelines for dynamic shapes; in the meantime it sometimes is faster if I just fix these things. :)